pure_hashget example fails

Hello,

bellow is the trace of command:

./pepper_compile_and_setup_V.sh pure_hashget pure_hashget.vkey pure_hashget.pkey
+ compiling common objs

============================================
=== Compiling computation to constraints ===
============================================

+ compile apps/pure_hashget.c --metrics -b 0 -w 10240 -t ZAATAR -db-hash-func ggh -db-num-addresses 1024 -ram-cell-num-bits 32 -db-hash-num-bits 1216 -db-thr-num-addresses-naive 32768 -fast-ram-address-width 32 -fast-ram-word-width 64
metric_num_lines_in_sfdl pure_hashget 273
metric_num_lines_in_source pure_hashget 273
make[1]: Entering directory '/home/fara/Downloads/pequin-master/compiler/frontend'
ant compile
Buildfile: /home/fara/Downloads/pequin-master/compiler/frontend/build.xml

compile:

BUILD SUCCESSFUL
Total time: 0 seconds
make[1]: Leaving directory '/home/fara/Downloads/pequin-master/compiler/frontend'
WARNING: --cstdarithtruncate is disabled, so type errors will warn and arithmetic is not ANSI C compliant
Compiling pure_hashget.c
Expanding circuit to file pure_hashget.c.ZAATAR.circuit
WARNING: Compiler may allow multiple assignments to 'const' variable with type const int
WARNING: Compiler may allow multiple assignments to 'const' variable with type const int
WARNING: Compiler may allow multiple assignments to 'const' variable with type const int
Writing constant values to pure_hashget.c.ZAATAR.spec.cons
metric_compile_1_utime pure_hashget 0.73
metric_compile_1_stime pure_hashget 0.05

metric_compile_1_wires pure_hashget 1159
metric_compile_tac2_utime pure_hashget 0.00
metric_compile_tac2_stime pure_hashget 0.00

WARNING: --cstdarithtruncate is disabled, so type errors will warn and arithmetic is not ANSI C compliant
metric_compile_p2_utime pure_hashget 0.28
metric_compile_p2_stime pure_hashget 0.05

metric_compile_untac2_utime pure_hashget 0.00
metric_compile_untac2_stime pure_hashget 0.00

WARNING: --cstdarithtruncate is disabled, so type errors will warn and arithmetic is not ANSI C compliant
Optimizing previously compiled circuit pure_hashget.c.ZAATAR.circuit2
Using variable dependencies from profile pure_hashget.c.ZAATAR.circuit2.profile
(Complete.)
Expanding circuit to file pure_hashget.c.ZAATAR.circuit
Writing constraints to pure_hashget.c.ZAATAR.spec_tmp
metric_num_occurrences_exo_compute 0
metric_num_occurrences_ext_gadget 0
metric_num_occurrences_hashget 2
metric_num_occurrences_hashput 0
metric_num_occurrences_ramget 0
metric_num_occurrences_ramget_fast 0
metric_num_occurrences_ramput 0
metric_num_occurrences_ramput_fast 0
Cleaning up constraints, result will appear in pure_hashget.c.ZAATAR.spec
metric_compile_2_utime pure_hashget 0.87
metric_compile_2_stime pure_hashget 0.09

metric_compile_2_wires pure_hashget 83
Expanding database operations in ../pure_hashget.c.ZAATAR.spec
Creating prover worksheet, result will appear at bin/pure_hashget.pws
Writing QAP matrices to ../../pepper/bin/pure_hashget.qap
Traceback (most recent call last):
  File "zcc_backend.py", line 331, in <module>
    main()
  File "zcc_backend.py", line 328, in main
    gen.generate_code_from_template(opt.spec)
  File "zcc_backend.py", line 209, in generate_code_from_template
    self.generate_matrices(spec_file, defs)
  File "zcc_backend.py", line 130, in generate_matrices
    (defs['NzA'], defs['NzB'], defs['NzC'], defs['num_constraints']) = zcc_parser.generate_zaatar_matrices(spec_file, shuffledIndices, qap_file_name)
  File "/home/fara/Downloads/pequin-master/compiler/backend/zcc_parser.py", line 1765, in generate_zaatar_matrices
    process_spec_section(spec_file, START_TAG + CONSTRAINTS_TAG, END_TAG + CONSTRAINTS_TAG, f)
  File "/home/fara/Downloads/pequin-master/compiler/backend/zcc_parser.py", line 81, in process_spec_section
    func(line)
  File "/home/fara/Downloads/pequin-master/compiler/backend/zcc_parser.py", line 1723, in f
    for bc in basic_constraints:
  File "/home/fara/Downloads/pequin-master/compiler/backend/zcc_parser.py", line 1251, in to_basic_constraints
    yield expand_basic_constraint(bc)
  File "/home/fara/Downloads/pequin-master/compiler/backend/zcc_parser.py", line 1266, in expand_basic_constraint
    expansion += expand_polynomial_str(tokens)
  File "/home/fara/Downloads/pequin-master/compiler/backend/zcc_parser.py", line 1280, in expand_polynomial_str
    expanded = expand_polynomial(tokens)
  File "/home/fara/Downloads/pequin-master/compiler/backend/zcc_parser.py", line 1294, in expand_polynomial
    raise Exception("expand_polynomial: Format error: " + str(tokens))
Exception: expand_polynomial: Format error: deque(['EXTERN', 'state_op_get_block_by_hash_4_ggh_64', '0'])
Makefile:113: recipe for target 'bin/pure_hashget.params' failed
make: *** [bin/pure_hashget.params] Error 1

==========================================
===== Running setup (key generation) =====
==========================================

./pepper_compile_and_setup_V.sh: line 35: bin/pepper_verifier_pure_hashget: No such file or directory

I have the same issue with genome_snp_freq.c or tolling.c. 

I also have a question, I noticed that there is not any example with float datatype. Is it possible to write a c code woth float datatype?

Thank you vey much,
Fattaneh

--- a/compiler/backend/zcc_parser.py
+++ b/compiler/backend/zcc_parser.py
@@ -1676,7 +1676,7 @@ def generate_zaatar_matrices(spec_file, shuffled_indices, qap_file_name):
       NzA += cons_entry.Aij
       NzB += cons_entry.Bij
       NzC += cons_entry.Cij
-    if line.startswith("EXT_GADGET"):
+    elif line.startswith("EXT_GADGET"):
       (varList, gadgetId) = parse_ext_gadget_pws_line(line)
 
       # Try calling into gadget binary

